File: backend/src/main/java/com/ngv/libraryManagementSystem/service/member/MemberService.java
package com.ngv.libraryManagementSystem.service.member;

import com.ngv.libraryManagementSystem.dto.response.MemberResponse;
import java.util.List;

public interface MemberService {
    MemberResponse getMemberById(Long id);
    MemberResponse getMyProfile(Long memberId);
    List<MemberResponse> getAllMembers();
    List<MemberResponse> getMembersByBookId(Long bookId);
}

File: frontend/src/app/admin/services/member.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiResponse } from '../../auth/dto/api-response';

export interface MemberResponse {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  joinDate: string;
  status: string;
}

@Injectable({
  providedIn: 'root',
})
export class MemberService {
  private readonly baseUrl = 'http://localhost:8080/api/members';

  constructor(private http: HttpClient) {}

  getAllMembers(): Observable<ApiResponse<MemberResponse[]>> {
    return this.http.get<ApiResponse<MemberResponse[]>>(`${this.baseUrl}`);
  }

  getMemberById(id: number): Observable<ApiResponse<MemberResponse>> {
    return this.http.get<ApiResponse<MemberResponse>>(`${this.baseUrl}/${id}`);
  }

  getMyProfile(): Observable<ApiResponse<MemberResponse>> {
    return this.http.get<ApiResponse<MemberResponse>>(`${this.baseUrl}/profile`);
  }
}

